### ISSUE TITLE
Overriding vhost-templates global

### ISSUE THREAD

**Issue by mavimedia** [cytopia/devilbox#144]
Is it possible to override the vhost-templates global, so it will be used for all projects?

**Comment by cytopia** [comment]
At the moment you can only alter general server settings. For that see `cfg/apache-2.4` directory which also includes an example.

Is this sufficient or do you need to alter the vhost itself in general?
If so, could you give some examples about what settings you require.

**Comment by mavimedia** [comment]
One can override the vhost-templates for each project like you described in the issue https://github.com/cytopia/devilbox/issues/142

I would like to do something like that, but global, so i don't need to put .devilbox folder inside the projects folder.

**Comment by cytopia** [comment]
As said, this is currently not possible. Depending on the options:

1. If yours are better default option than current present, I will add them to: https://github.com/devilbox/vhost-gen
2. If yours are very specific to your local dev environment I will have to implement a way to generally overwrite the defaults

**Comment by mavimedia** [comment]
I think, it would be nice, if all projects are also accessible over www.project.loc, not only over project.loc. But maybe other users don't want this behaviour like me.

**Comment by cytopia** [comment]
@mavimedia OK.

So this is not (yet) officially ported, but as a quick work-around you can do the following:

---

1. In the devilbox root directory, create a new directory `vhost-gen/`
2. Copy all three templates files into the new `vhost-gen/` dir from: https://github.com/devilbox/vhost-gen/tree/master/etc/templates
3. Add a new volume mount as shown below into `docker-compose.yml`

```yml
  httpd:
    ...
    volumes:
      ...
      - ${DEVILBOX_PATH}/templates:/etc/vhost-gen/templates
      ...
```

This way you should be able to edit the templates globally for all vhosts from within the devilbox project.

Please let me know how this works out for you.

**Comment by mavimedia** [comment]
It works, thanks!

But in ypur example it should be

  httpd:
    ...
    volumes:
      ...
      - ${DEVILBOX_PATH}/**vhost-gen**:/etc/vhost-gen/templates
      ...

**Comment by cytopia** [comment]
Yes you are right. I will fix my comment above for others to better see how it works.

**Comment by cytopia** [comment]
The whole thing is going to be documented with new documentation setup: https://github.com/cytopia/devilbox/issues/157

**Comment by cytopia** [comment]
This will be fixed and documented here:
https://github.com/cytopia/devilbox/pull/298

**Comment by cytopia** [comment]
@mavimedia feature will be very soon fully integrated and documented

**Comment by keithy** [comment]
Not working for me.

**Comment by cytopia** [comment]
Good catch! Should be fixed by the above listed PR.
